NVDARemote Menu Incorrectly Shows 'Disconnect' After Failed Connection Attempt to Non-Existent Server

NVDARemote menu incorrectly shows 'Disconnect' after failed connection attempt to non-existent server

### A Brief Description
When attempting to connect to a non-existent or offline NVDARemote server (in slave mode and master mode (while waiting for the NVDARemote to connected to server)), the Remote Access menu (NVDA Menu > Tools > Remote Access) incorrectly remains set to "Disconnect" even though no active connection exists.
#### After Triggering a Failed Connection
1. The menu wrongly displays "Disconnect" instead of reverting to "Connect".
2. Selecting "Disconnect" will act as usual (show the disconnect confirmation, play the sound and NVDA announce "disconnect") but the menu state does not update.
3. If you select the "disconnect" for the second time or more, it will do nothing until you recover this issue.
4. The only recovery methods are using the hotkey to connect to a valid server (then disconnecting properly) or restarting NVDA entirely.
#### Impact
Users cannot reconnect via the menu after a failed attempt, forcing reliance on hotkeys or NVDA restarts.

### Steps to Reproduce
1. Ensure no NVDARemote server is running or available.
2. Go to "NVDA Menu > Tools > Remote Access > Connect" or use the NVDARemote hotkey to connect.
3. In the connection dialog, enter details for a non-existent server (or one that isn't running) and attempt to connect in slave mode.
4. Observe that the connection fails (expected), but the menu still shows "Disconnect" instead of "Connect".
5. If you using hotkey in this state, the connect dialog appears.
6. Select "Disconnect" from the menu and confirm the disconnection.
7. NVDA announces "Disconnected" and plays the disconnect sound, but the menu still shows "Disconnect".
8. Select "Disconnect" from the menu again do nothing.
9. The only ways to recover are to use the hotkey to connect to a valid server or restart NVDA.

#### Note
This also occurs in master mode during the connection attempt phase, but is less noticeable because successful connections often happen quickly.

### Actual Behavior
- After failing to connect to a non-existent server, the Remote Access menu incorrectly shows "Disconnect" instead of "Connect".
- Attempting to disconnect from this non-existent connection for the first time shows the disconnect confirmation dialog and plays the disconnect sound, but doesn't correct the menu state. 
- Attempting to disconnect from this non-existent connection for the second time or more do nothing.
- The menu remains stuck showing "Disconnect" until either successfully connecting to a valid server using the hotkey and then disconnecting or restarting NVDA

### Expected Behavior
- After failing to connect to a server, the Remote Access menu should show "Connect" (not "Disconnect").
- Attempting to disconnect when not actually connected should either not show the disconnect option at all, or show the option but have it correctly return to "Connect" state after selection.

### Debug Log
```
IO - inputCore.InputManager.executeGesture (14:13:51.984) - winInputHook (12660):
Input: kb(laptop):NVDA+n
IO - speech.speech.speak (14:13:52.026) - MainThread (2936):
Speaking ['NVDA', 'menu', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:13:52.263) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:13:52.269) - MainThread (2936):
Speaking ['Preferences', 'subMenu', CharacterModeCommand(True), 'p', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:13:52.428) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:13:52.434) - MainThread (2936):
Speaking ['Tools', 'subMenu', CharacterModeCommand(True), 't', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:13:53.326) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:13:53.339) - MainThread (2936):
Speaking ['View log', CharacterModeCommand(True), 'l', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:13:53.855) - winInputHook (12660):
Input: kb(laptop):upArrow
IO - speech.speech.speak (14:13:53.862) - MainThread (2936):
Speaking ['Remote Access', 'subMenu', CharacterModeCommand(True), 'e', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:13:54.657) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:13:54.669) - MainThread (2936):
Speaking ['Connect...', CharacterModeCommand(True), 'c', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:13:55.271) - winInputHook (12660):
Input: kb(laptop):enter
DEBUG - gui.contextHelp.bindHelpEvent (14:13:55.279) - MainThread (2936):
Did context help binding for DirectConnectDialog
DEBUG - gui.contextHelp.bindHelpEvent (14:13:55.294) - MainThread (2936):
Did context help binding for ClientPanel
DEBUG - gui.contextHelp.bindHelpEvent (14:13:55.310) - MainThread (2936):
Did context help binding for ServerPanel
DEBUGWARNING - Python warning (14:13:55.342) - MainThread (2936):
gui\__init__.pyc:910: DeprecationWarning: showScriptModalDialog is deprecated. Use an instance of message.MessageDialog and wx.CallAfter instead.
IO - speech.speech.speak (14:13:55.443) - MainThread (2936):
Speaking ['Connect to Another Computer', 'dialog', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:13:55.449) - MainThread (2936):
Speaking ['Mode:', 'combo box', 'Allow this computer to be controlled', 'collapsed', 'Alt+', CharacterModeCommand(True), 'm', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:13:58.060) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:13:58.070) - MainThread (2936):
Speaking ['Server:', 'combo box', 'Use existing', 'collapsed', 'Alt+', CharacterModeCommand(True), 's', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:13:58.307) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:13:58.325) - MainThread (2936):
Speaking ['Host:', 'combo box', 'collapsed', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:13:58.326) - MainThread (2936):
Speaking ['edit', 'Alt+', CharacterModeCommand(True), 'h', CharacterModeCommand(False), CancellableSpeech (still valid), 'selected myownserver.com:6837']
IO - inputCore.InputManager.executeGesture (14:13:58.717) - winInputHook (12660):
Input: kb(laptop):1
IO - speech.speech.speak (14:13:58.724) - MainThread (2936):
Speaking [CharacterModeCommand(True), '1', EndUtteranceCommand()]
IO - speech.speech.speak (14:13:58.726) - MainThread (2936):
Speaking ['selection removed']
IO - inputCore.InputManager.executeGesture (14:13:58.900) - winInputHook (12660):
Input: kb(laptop):2
IO - speech.speech.speak (14:13:58.925) - MainThread (2936):
Speaking [CharacterModeCommand(True), '2', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:13:59.090) - winInputHook (12660):
Input: kb(laptop):3
IO - speech.speech.speak (14:13:59.109) - MainThread (2936):
Speaking [CharacterModeCommand(True), '3', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:13:59.339) - winInputHook (12660):
Input: kb(laptop):.
IO - speech.speech.speakTypedCharacters (14:13:59.355) - MainThread (2936):
typed word: 123
IO - speech.speech.speak (14:13:59.357) - MainThread (2936):
Speaking ['dot', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:13:59.525) - winInputHook (12660):
Input: kb(laptop):c
IO - speech.speech.speak (14:13:59.545) - MainThread (2936):
Speaking [CharacterModeCommand(True), 'c', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:13:59.681) - winInputHook (12660):
Input: kb(laptop):o
IO - speech.speech.speak (14:13:59.696) - MainThread (2936):
Speaking [CharacterModeCommand(True), 'o', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:13:59.885) - winInputHook (12660):
Input: kb(laptop):m
IO - speech.speech.speak (14:13:59.900) - MainThread (2936):
Speaking [CharacterModeCommand(True), 'm', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:14:00.057) - winInputHook (12660):
Input: kb(laptop):tab
DEBUG - NVDAObjects.NVDAObject._get_placeholder (14:14:00.069) - MainThread (2936):
Potential unimplemented child class: <NVDAObjects.Dynamic_IAccessibleEditWindowNVDAObject object at 0x0974A7F0>
IO - speech.speech.speak (14:14:00.069) - MainThread (2936):
Speaking ['Key:', 'edit', 'Alt+', CharacterModeCommand(True), 'k', CharacterModeCommand(False), CancellableSpeech (still valid), 'blank']
DEBUG - utils.schedule.callJobOnThread (14:14:00.145) - ScheduleThread (2260):
Starting thread for job: _checkForUpdatableAddons on thread None
DEBUG - gui.addonStoreGui.controls.messageDialogs.UpdatableAddonsDialog._checkForUpdatableAddons (14:14:00.145) - _checkForUpdatableAddons (15628):
checking for updatable add-ons
DEBUG - addonStore.dataManager._DataManager._getCacheHash (14:14:00.145) - _checkForUpdatableAddons (15628):
Fetching add-on data from https://addonStore.nvaccess.org/cacheHash.json
DEBUG - gui.addonStoreGui.controls.messageDialogs.UpdatableAddonsDialog._checkForUpdatableAddons (14:14:00.259) - _checkForUpdatableAddons (15628):
no updatable add-ons found
IO - inputCore.InputManager.executeGesture (14:14:00.982) - winInputHook (12660):
Input: kb(laptop):4
IO - speech.speech.speak (14:14:00.998) - MainThread (2936):
Speaking [CharacterModeCommand(True), '4', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:14:01.176) - winInputHook (12660):
Input: kb(laptop):5
IO - speech.speech.speak (14:14:01.201) - MainThread (2936):
Speaking [CharacterModeCommand(True), '5', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:14:01.393) - winInputHook (12660):
Input: kb(laptop):6
IO - speech.speech.speak (14:14:01.397) - MainThread (2936):
Speaking [CharacterModeCommand(True), '6', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:14:02.253) - winInputHook (12660):
Input: kb(laptop):enter
INFO - _remoteClient.client.RemoteClient.connect (14:14:02.268) - MainThread (2936):
Initiating connection as slave to 123.com:6837
INFO - _remoteClient.transport.RelayTransport.__init__ (14:14:02.268) - MainThread (2936):
Connecting to ('123.com', 6837) channel 456
INFO - _remoteClient.session.RemoteSession.__init__ (14:14:02.268) - MainThread (2936):
Initializing Remote Session
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.269) - MainThread (2936):
Creating new handler for version_mismatch
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.269) - MainThread (2936):
Registering handler for version_mismatch
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.269) - MainThread (2936):
Creating new handler for motd
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.269) - MainThread (2936):
Registering handler for motd
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.269) - MainThread (2936):
Creating new handler for set_clipboard_text
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.270) - MainThread (2936):
Registering handler for set_clipboard_text
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.270) - MainThread (2936):
Creating new handler for client_joined
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.270) - MainThread (2936):
Registering handler for client_joined
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.270) - MainThread (2936):
Creating new handler for client_left
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.270) - MainThread (2936):
Registering handler for client_left
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.270) - MainThread (2936):
Creating new handler for key
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.270) - MainThread (2936):
Registering handler for key
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.270) - MainThread (2936):
Creating new handler for channel_joined
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.272) - MainThread (2936):
Registering handler for channel_joined
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.272) - MainThread (2936):
Creating new handler for set_braille_info
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.272) - MainThread (2936):
Registering handler for set_braille_info
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.272) - MainThread (2936):
Creating new handler for set_display_size
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.272) - MainThread (2936):
Registering handler for set_display_size
WARNING - extensionPoints.util.HandlerRegistrar.register (14:14:02.273) - MainThread (2936):
braille.filter_displaySize is deprecated. Use braille.filter_displayDimensions instead.
Stack trace:
  File "nvda.pyw", line 309, in <module>
  File "core.pyc", line 1049, in main
  File "wx\core.pyc", line 2262, in MainLoop
  File "wx\core.pyc", line 3427, in <lambda>
  File "gui\__init__.pyc", line 918, in run
  File "_remoteClient\client.pyc", line 318, in handleDialogCompletion
  File "_remoteClient\client.pyc", line 206, in connect
  File "_remoteClient\client.pyc", line 376, in connectAsFollower
  File "_remoteClient\session.pyc", line 318, in __init__
  File "extensionPoints\util.pyc", line 129, in register
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.273) - MainThread (2936):
Creating new handler for braille_input
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.274) - MainThread (2936):
Registering handler for braille_input
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.274) - MainThread (2936):
Creating new handler for send_SAS
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.274) - MainThread (2936):
Registering handler for send_SAS
INFO - _remoteClient.secureDesktop.SecureDesktopHandler.followerSession (14:14:02.274) - MainThread (2936):
Updating follower session reference
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:02.274) - MainThread (2936):
Registering handler for set_braille_info
DEBUGWARNING - Python warning (14:14:02.291) - Thread-25_connector_loop (15892):
_remoteClient\transport.pyc:436: DeprecationWarning: ssl.PROTOCOL_TLSv1_2 is deprecated
IO - speech.speech.speak (14:14:02.355) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:02.360) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
IO - inputCore.InputManager.executeGesture (14:14:06.620) - winInputHook (12660):
Input: kb(laptop):NVDA+alt+r
DEBUG - gui.contextHelp.bindHelpEvent (14:14:06.625) - MainThread (2936):
Did context help binding for DirectConnectDialog
DEBUG - gui.contextHelp.bindHelpEvent (14:14:06.674) - MainThread (2936):
Did context help binding for ClientPanel
DEBUG - gui.contextHelp.bindHelpEvent (14:14:06.712) - MainThread (2936):
Did context help binding for ServerPanel
IO - speech.speech.speak (14:14:06.876) - MainThread (2936):
Speaking ['Connect to Another Computer', 'dialog', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:06.878) - MainThread (2936):
Speaking ['Mode:', 'combo box', 'Allow this computer to be controlled', 'collapsed', 'Alt+', CharacterModeCommand(True), 'm', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:08.233) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:08.242) - MainThread (2936):
Speaking ['Server:', 'combo box', 'Use existing', 'collapsed', 'Alt+', CharacterModeCommand(True), 's', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:08.491) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:08.508) - MainThread (2936):
Speaking ['Host:', 'combo box', 'collapsed', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:08.509) - MainThread (2936):
Speaking ['edit', 'Alt+', CharacterModeCommand(True), 'h', CharacterModeCommand(False), CancellableSpeech (still valid), 'selected myownserver.com:6837']
IO - inputCore.InputManager.executeGesture (14:14:08.781) - winInputHook (12660):
Input: kb(laptop):tab
DEBUG - NVDAObjects.NVDAObject._get_placeholder (14:14:08.796) - MainThread (2936):
Potential unimplemented child class: <NVDAObjects.Dynamic_IAccessibleEditWindowNVDAObject object at 0x09744950>
IO - speech.speech.speak (14:14:08.796) - MainThread (2936):
Speaking ['Key:', 'edit', 'Alt+', CharacterModeCommand(True), 'k', CharacterModeCommand(False), CancellableSpeech (still valid), 'blank']
IO - inputCore.InputManager.executeGesture (14:14:09.541) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:09.553) - MainThread (2936):
Speaking ['Generate Key', 'button', 'Alt+', CharacterModeCommand(True), 'g', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:09.962) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:09.975) - MainThread (2936):
Speaking ['OK', 'button', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:10.298) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:10.306) - MainThread (2936):
Speaking ['Cancel', 'button', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:10.716) - winInputHook (12660):
Input: kb(laptop):space
IO - speech.speech.speak (14:14:10.731) - MainThread (2936):
Speaking ['space', EndUtteranceCommand()]
IO - speech.speech.speak (14:14:10.731) - MainThread (2936):
Speaking ['pressed']
IO - speech.speech.speak (14:14:10.819) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:10.825) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
IO - inputCore.InputManager.executeGesture (14:14:11.281) - winInputHook (12660):
Input: kb(laptop):NVDA+n
DEBUG - gui.SysTrayIcon.evaluateUpdatePendingUpdateMenuItemCommand (14:14:11.282) - MainThread (2936):
Error while removing pending update menu item
Traceback (most recent call last):
  File "gui\__init__.pyc", line 720, in evaluateUpdatePendingUpdateMenuItemCommand
wx._core.wxAssertionError: C++ assertion ""node"" failed at ..\..\src\common\menucmn.cpp(462) in wxMenuBase::Remove(): removing item not in the menu?
IO - speech.speech.speak (14:14:11.313) - MainThread (2936):
Speaking ['NVDA', 'menu', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:11.452) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:11.457) - MainThread (2936):
Speaking ['Preferences', 'subMenu', CharacterModeCommand(True), 'p', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:11.634) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:11.641) - MainThread (2936):
Speaking ['Tools', 'subMenu', CharacterModeCommand(True), 't', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:11.834) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:11.848) - MainThread (2936):
Speaking ['View log', CharacterModeCommand(True), 'l', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:12.050) - winInputHook (12660):
Input: kb(laptop):upArrow
IO - speech.speech.speak (14:14:12.058) - MainThread (2936):
Speaking ['Remote Access', 'subMenu', CharacterModeCommand(True), 'e', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:12.213) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:12.227) - MainThread (2936):
Speaking ['Disconnect', CharacterModeCommand(True), 'd', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:12.762) - winInputHook (12660):
Input: kb(laptop):enter
DEBUG - gui.contextHelp.bindHelpEvent (14:14:12.783) - MainThread (2936):
Did context help binding for MessageDialog
DEBUG - gui.message.MessageDialog.ShowModal (14:14:12.803) - MainThread (2936):
Adding <gui.message.MessageDialog object at 0x0994FDF0> to instances.
DEBUG - gui.message.MessageDialog.ShowModal (14:14:12.803) - MainThread (2936):
Showing <gui.message.MessageDialog object at 0x0994FDF0> as modal
IO - speech.speech.speak (14:14:12.864) - MainThread (2936):
Speaking ['Confirm Disconnection', 'dialog', 'Are you sure you want to disconnect from the Remote Access session?', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:12.870) - MainThread (2936):
Speaking ['No', 'button', 'Alt+', CharacterModeCommand(True), 'n', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:15.677) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:15.683) - MainThread (2936):
Speaking ['Yes', 'button', 'Alt+', CharacterModeCommand(True), 'y', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:16.413) - winInputHook (12660):
Input: kb(laptop):space
IO - speech.speech.speak (14:14:16.431) - MainThread (2936):
Speaking ['space', EndUtteranceCommand()]
IO - speech.speech.speak (14:14:16.433) - MainThread (2936):
Speaking ['pressed']
DEBUG - gui.message.MessageDialog._onButtonEvent (14:14:16.446) - MainThread (2936):
Got button event on id=5103
DEBUG - gui.message.MessageDialog._onCloseEvent (14:14:16.459) - MainThread (2936):
Queueing <gui.message.MessageDialog object at 0x0994FDF0> for destruction
DEBUG - gui.message.MessageDialog._onCloseEvent (14:14:16.459) - MainThread (2936):
Removing <gui.message.MessageDialog object at 0x0994FDF0> from instances.
INFO - _remoteClient.client.RemoteClient.disconnect (14:14:16.459) - MainThread (2936):
Disconnecting from remote session
INFO - _remoteClient.secureDesktop.SecureDesktopHandler.followerSession (14:14:16.460) - MainThread (2936):
Updating follower session reference
DEBUG - _remoteClient.transport.Transport.unregisterInbound (14:14:16.460) - MainThread (2936):
Unregistered handler for set_braille_info
IO - speech.speech.speak (14:14:16.719) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:16.725) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
DEBUG - speech.manager.SpeechManager._handleIndex (14:14:16.726) - MainThread (2936):
Unknown index 51, speech probably cancelled from main thread.
IO - speech.speech.speak (14:14:16.739) - MainThread (2936):
Speaking ['Disconnected']
IO - inputCore.InputManager.executeGesture (14:14:17.956) - winInputHook (12660):
Input: kb(laptop):NVDA+n
DEBUG - gui.SysTrayIcon.evaluateUpdatePendingUpdateMenuItemCommand (14:14:17.959) - MainThread (2936):
Error while removing pending update menu item
Traceback (most recent call last):
  File "gui\__init__.pyc", line 720, in evaluateUpdatePendingUpdateMenuItemCommand
wx._core.wxAssertionError: C++ assertion ""node"" failed at ..\..\src\common\menucmn.cpp(462) in wxMenuBase::Remove(): removing item not in the menu?
IO - speech.speech.speak (14:14:17.987) - MainThread (2936):
Speaking ['NVDA', 'menu', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:18.110) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:18.116) - MainThread (2936):
Speaking ['Preferences', 'subMenu', CharacterModeCommand(True), 'p', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:18.263) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:18.269) - MainThread (2936):
Speaking ['Tools', 'subMenu', CharacterModeCommand(True), 't', CharacterModeCommand(False), CancellableSpeech (still valid)]
INFO - _remoteClient.transport.ConnectorThread.run (14:14:18.281) - Thread-25_connector_loop (15892):
Ending control connector thread Thread-25_connector_loop
IO - inputCore.InputManager.executeGesture (14:14:18.412) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:18.426) - MainThread (2936):
Speaking ['View log', CharacterModeCommand(True), 'l', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:18.590) - winInputHook (12660):
Input: kb(laptop):upArrow
IO - speech.speech.speak (14:14:18.596) - MainThread (2936):
Speaking ['Remote Access', 'subMenu', CharacterModeCommand(True), 'e', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:18.886) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:18.902) - MainThread (2936):
Speaking ['Disconnect', CharacterModeCommand(True), 'd', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:19.289) - winInputHook (12660):
Input: kb(laptop):enter
DEBUG - _remoteClient.client.RemoteClient.disconnect (14:14:19.308) - MainThread (2936):
Disconnect called but no active sessions
IO - speech.speech.speak (14:14:19.330) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:19.344) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
IO - inputCore.InputManager.executeGesture (14:14:19.831) - winInputHook (12660):
Input: kb(laptop):NVDA+n
DEBUG - gui.SysTrayIcon.evaluateUpdatePendingUpdateMenuItemCommand (14:14:19.832) - MainThread (2936):
Error while removing pending update menu item
Traceback (most recent call last):
  File "gui\__init__.pyc", line 720, in evaluateUpdatePendingUpdateMenuItemCommand
wx._core.wxAssertionError: C++ assertion ""node"" failed at ..\..\src\common\menucmn.cpp(462) in wxMenuBase::Remove(): removing item not in the menu?
IO - speech.speech.speak (14:14:19.864) - MainThread (2936):
Speaking ['NVDA', 'menu', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:20.036) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:20.041) - MainThread (2936):
Speaking ['Preferences', 'subMenu', CharacterModeCommand(True), 'p', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:20.199) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:20.205) - MainThread (2936):
Speaking ['Tools', 'subMenu', CharacterModeCommand(True), 't', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:20.384) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:20.399) - MainThread (2936):
Speaking ['View log', CharacterModeCommand(True), 'l', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:20.563) - winInputHook (12660):
Input: kb(laptop):upArrow
IO - speech.speech.speak (14:14:20.569) - MainThread (2936):
Speaking ['Remote Access', 'subMenu', CharacterModeCommand(True), 'e', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:20.735) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:20.750) - MainThread (2936):
Speaking ['Disconnect', CharacterModeCommand(True), 'd', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:21.762) - winInputHook (12660):
Input: kb(laptop):enter
DEBUG - _remoteClient.client.RemoteClient.disconnect (14:14:21.779) - MainThread (2936):
Disconnect called but no active sessions
IO - speech.speech.speak (14:14:21.801) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:21.816) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
IO - inputCore.InputManager.executeGesture (14:14:22.395) - winInputHook (12660):
Input: kb(laptop):NVDA+alt+r
DEBUG - gui.contextHelp.bindHelpEvent (14:14:22.399) - MainThread (2936):
Did context help binding for DirectConnectDialog
DEBUG - gui.contextHelp.bindHelpEvent (14:14:22.414) - MainThread (2936):
Did context help binding for ClientPanel
DEBUG - gui.contextHelp.bindHelpEvent (14:14:22.429) - MainThread (2936):
Did context help binding for ServerPanel
IO - speech.speech.speak (14:14:22.546) - MainThread (2936):
Speaking ['Connect to Another Computer', 'dialog', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:22.548) - MainThread (2936):
Speaking ['Mode:', 'combo box', 'Allow this computer to be controlled', 'collapsed', 'Alt+', CharacterModeCommand(True), 'm', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:23.200) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:23.208) - MainThread (2936):
Speaking ['Server:', 'combo box', 'Use existing', 'collapsed', 'Alt+', CharacterModeCommand(True), 's', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:23.456) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:23.476) - MainThread (2936):
Speaking ['Host:', 'combo box', 'collapsed', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:23.477) - MainThread (2936):
Speaking ['edit', 'Alt+', CharacterModeCommand(True), 'h', CharacterModeCommand(False), CancellableSpeech (still valid), 'selected myownserver.com:6837']
IO - inputCore.InputManager.executeGesture (14:14:23.691) - winInputHook (12660):
Input: kb(laptop):tab
DEBUG - NVDAObjects.NVDAObject._get_placeholder (14:14:23.703) - MainThread (2936):
Potential unimplemented child class: <NVDAObjects.Dynamic_IAccessibleEditWindowNVDAObject object at 0x09723FD0>
IO - speech.speech.speak (14:14:23.703) - MainThread (2936):
Speaking ['Key:', 'edit', 'Alt+', CharacterModeCommand(True), 'k', CharacterModeCommand(False), CancellableSpeech (still valid), 'blank']
IO - inputCore.InputManager.executeGesture (14:14:23.906) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:23.918) - MainThread (2936):
Speaking ['Generate Key', 'button', 'Alt+', CharacterModeCommand(True), 'g', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:24.122) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:24.130) - MainThread (2936):
Speaking ['OK', 'button', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:24.335) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:24.343) - MainThread (2936):
Speaking ['Cancel', 'button', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:24.601) - winInputHook (12660):
Input: kb(laptop):space
IO - speech.speech.speak (14:14:24.622) - MainThread (2936):
Speaking ['space', EndUtteranceCommand()]
IO - speech.speech.speak (14:14:24.623) - MainThread (2936):
Speaking ['pressed']
IO - speech.speech.speak (14:14:24.738) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:24.747) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
IO - inputCore.InputManager.executeGesture (14:14:25.087) - winInputHook (12660):
Input: kb(laptop):NVDA+n
DEBUG - gui.SysTrayIcon.evaluateUpdatePendingUpdateMenuItemCommand (14:14:25.087) - MainThread (2936):
Error while removing pending update menu item
Traceback (most recent call last):
  File "gui\__init__.pyc", line 720, in evaluateUpdatePendingUpdateMenuItemCommand
wx._core.wxAssertionError: C++ assertion ""node"" failed at ..\..\src\common\menucmn.cpp(462) in wxMenuBase::Remove(): removing item not in the menu?
IO - speech.speech.speak (14:14:25.120) - MainThread (2936):
Speaking ['NVDA', 'menu', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:25.280) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:25.285) - MainThread (2936):
Speaking ['Preferences', 'subMenu', CharacterModeCommand(True), 'p', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:25.441) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:25.447) - MainThread (2936):
Speaking ['Tools', 'subMenu', CharacterModeCommand(True), 't', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:25.633) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:25.646) - MainThread (2936):
Speaking ['View log', CharacterModeCommand(True), 'l', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:25.804) - winInputHook (12660):
Input: kb(laptop):upArrow
IO - speech.speech.speak (14:14:25.809) - MainThread (2936):
Speaking ['Remote Access', 'subMenu', CharacterModeCommand(True), 'e', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:25.956) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:25.969) - MainThread (2936):
Speaking ['Disconnect', CharacterModeCommand(True), 'd', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:26.374) - winInputHook (12660):
Input: kb(laptop):enter
DEBUG - _remoteClient.client.RemoteClient.disconnect (14:14:26.393) - MainThread (2936):
Disconnect called but no active sessions
IO - speech.speech.speak (14:14:26.413) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:26.427) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
IO - inputCore.InputManager.executeGesture (14:14:27.670) - winInputHook (12660):
Input: kb(laptop):NVDA+alt+r
DEBUG - gui.contextHelp.bindHelpEvent (14:14:27.673) - MainThread (2936):
Did context help binding for DirectConnectDialog
DEBUG - gui.contextHelp.bindHelpEvent (14:14:27.688) - MainThread (2936):
Did context help binding for ClientPanel
DEBUG - gui.contextHelp.bindHelpEvent (14:14:27.703) - MainThread (2936):
Did context help binding for ServerPanel
IO - speech.speech.speak (14:14:27.823) - MainThread (2936):
Speaking ['Connect to Another Computer', 'dialog', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:27.826) - MainThread (2936):
Speaking ['Mode:', 'combo box', 'Allow this computer to be controlled', 'collapsed', 'Alt+', CharacterModeCommand(True), 'm', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:28.202) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:28.212) - MainThread (2936):
Speaking ['Server:', 'combo box', 'Use existing', 'collapsed', 'Alt+', CharacterModeCommand(True), 's', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:28.394) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:28.413) - MainThread (2936):
Speaking ['Host:', 'combo box', 'collapsed', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:28.415) - MainThread (2936):
Speaking ['edit', 'Alt+', CharacterModeCommand(True), 'h', CharacterModeCommand(False), CancellableSpeech (still valid), 'selected myownserver.com:6837']
IO - inputCore.InputManager.executeGesture (14:14:30.779) - winInputHook (12660):
Input: kb(laptop):tab
DEBUG - NVDAObjects.NVDAObject._get_placeholder (14:14:30.794) - MainThread (2936):
Potential unimplemented child class: <NVDAObjects.Dynamic_IAccessibleEditWindowNVDAObject object at 0x099205F0>
IO - speech.speech.speak (14:14:30.794) - MainThread (2936):
Speaking ['Key:', 'edit', 'Alt+', CharacterModeCommand(True), 'k', CharacterModeCommand(False), CancellableSpeech (still valid), 'blank']
IO - inputCore.InputManager.executeGesture (14:14:31.001) - winInputHook (12660):
Input: kb(laptop):1
IO - speech.speech.speak (14:14:31.004) - MainThread (2936):
Speaking [CharacterModeCommand(True), '1', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:14:31.231) - winInputHook (12660):
Input: kb(laptop):2
IO - speech.speech.speak (14:14:31.243) - MainThread (2936):
Speaking [CharacterModeCommand(True), '2', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:14:31.398) - winInputHook (12660):
Input: kb(laptop):3
IO - speech.speech.speak (14:14:31.416) - MainThread (2936):
Speaking [CharacterModeCommand(True), '3', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (14:14:31.524) - winInputHook (12660):
Input: kb(laptop):enter
INFO - _remoteClient.client.RemoteClient.connect (14:14:31.537) - MainThread (2936):
Initiating connection as slave to myownserver.com:6837
INFO - _remoteClient.transport.RelayTransport.__init__ (14:14:31.537) - MainThread (2936):
Connecting to ('myownserver.com', 6837) channel 123
INFO - _remoteClient.session.RemoteSession.__init__ (14:14:31.538) - MainThread (2936):
Initializing Remote Session
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.538) - MainThread (2936):
Creating new handler for version_mismatch
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.538) - MainThread (2936):
Registering handler for version_mismatch
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.538) - MainThread (2936):
Creating new handler for motd
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.538) - MainThread (2936):
Registering handler for motd
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.538) - MainThread (2936):
Creating new handler for set_clipboard_text
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.538) - MainThread (2936):
Registering handler for set_clipboard_text
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.538) - MainThread (2936):
Creating new handler for client_joined
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.538) - MainThread (2936):
Registering handler for client_joined
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.538) - MainThread (2936):
Creating new handler for client_left
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.539) - MainThread (2936):
Registering handler for client_left
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.539) - MainThread (2936):
Creating new handler for key
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.539) - MainThread (2936):
Registering handler for key
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.539) - MainThread (2936):
Creating new handler for channel_joined
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.539) - MainThread (2936):
Registering handler for channel_joined
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.539) - MainThread (2936):
Creating new handler for set_braille_info
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.539) - MainThread (2936):
Registering handler for set_braille_info
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.540) - MainThread (2936):
Creating new handler for set_display_size
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.540) - MainThread (2936):
Registering handler for set_display_size
WARNING - extensionPoints.util.HandlerRegistrar.register (14:14:31.541) - MainThread (2936):
braille.filter_displaySize is deprecated. Use braille.filter_displayDimensions instead.
Stack trace:
  File "nvda.pyw", line 309, in <module>
  File "core.pyc", line 1049, in main
  File "wx\core.pyc", line 2262, in MainLoop
  File "wx\core.pyc", line 3427, in <lambda>
  File "gui\__init__.pyc", line 918, in run
  File "_remoteClient\client.pyc", line 318, in handleDialogCompletion
  File "_remoteClient\client.pyc", line 206, in connect
  File "_remoteClient\client.pyc", line 376, in connectAsFollower
  File "_remoteClient\session.pyc", line 318, in __init__
  File "extensionPoints\util.pyc", line 129, in register
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.541) - MainThread (2936):
Creating new handler for braille_input
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.541) - MainThread (2936):
Registering handler for braille_input
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.541) - MainThread (2936):
Creating new handler for send_SAS
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.541) - MainThread (2936):
Registering handler for send_SAS
INFO - _remoteClient.secureDesktop.SecureDesktopHandler.followerSession (14:14:31.541) - MainThread (2936):
Updating follower session reference
DEBUG - _remoteClient.transport.Transport.registerInbound (14:14:31.541) - MainThread (2936):
Registering handler for set_braille_info
IO - speech.speech.speak (14:14:31.605) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:31.612) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
DEBUGWARNING - Python warning (14:14:31.625) - Thread-65_connector_loop (14556):
_remoteClient\transport.pyc:439: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
WARNING - logging.Logger.warn (14:14:31.625) - Thread-65_connector_loop (14556):
Skipping certificate verification for myownserver.com:6837
WARNING - logging.Logger.warn (14:14:31.742) - Thread-65_connector_loop (14556):
Skipping certificate verification for myownserver.com:6837
INFO - _remoteClient.client.onConnectedAsFollower (14:14:31.855) - MainThread (2936):
Control connector connected
IO - speech.speech.speak (14:14:32.159) - MainThread (2936):
Speaking ['Connected as controlled computer']
IO - inputCore.InputManager.executeGesture (14:14:33.723) - winInputHook (12660):
Input: kb(laptop):NVDA+n
DEBUG - gui.SysTrayIcon.evaluateUpdatePendingUpdateMenuItemCommand (14:14:33.724) - MainThread (2936):
Error while removing pending update menu item
Traceback (most recent call last):
  File "gui\__init__.pyc", line 720, in evaluateUpdatePendingUpdateMenuItemCommand
wx._core.wxAssertionError: C++ assertion ""node"" failed at ..\..\src\common\menucmn.cpp(462) in wxMenuBase::Remove(): removing item not in the menu?
IO - speech.speech.speak (14:14:33.753) - MainThread (2936):
Speaking ['NVDA', 'menu', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:33.871) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:33.876) - MainThread (2936):
Speaking ['Preferences', 'subMenu', CharacterModeCommand(True), 'p', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:34.028) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:34.034) - MainThread (2936):
Speaking ['Tools', 'subMenu', CharacterModeCommand(True), 't', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:34.205) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:34.219) - MainThread (2936):
Speaking ['View log', CharacterModeCommand(True), 'l', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:34.375) - winInputHook (12660):
Input: kb(laptop):upArrow
IO - speech.speech.speak (14:14:34.380) - MainThread (2936):
Speaking ['Remote Access', 'subMenu', CharacterModeCommand(True), 'e', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:34.542) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:34.554) - MainThread (2936):
Speaking ['Disconnect', CharacterModeCommand(True), 'd', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:35.376) - winInputHook (12660):
Input: kb(laptop):enter
DEBUG - gui.contextHelp.bindHelpEvent (14:14:35.399) - MainThread (2936):
Did context help binding for MessageDialog
DEBUG - gui.message.MessageDialog.ShowModal (14:14:35.415) - MainThread (2936):
Adding <gui.message.MessageDialog object at 0x040ABFA0> to instances.
DEBUG - gui.message.MessageDialog.ShowModal (14:14:35.415) - MainThread (2936):
Showing <gui.message.MessageDialog object at 0x040ABFA0> as modal
IO - speech.speech.speak (14:14:35.464) - MainThread (2936):
Speaking ['Confirm Disconnection', 'dialog', 'Are you sure you want to disconnect from the Remote Access session?', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:35.484) - MainThread (2936):
Speaking ['No', 'button', 'Alt+', CharacterModeCommand(True), 'n', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:36.346) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:36.354) - MainThread (2936):
Speaking ['Yes', 'button', 'Alt+', CharacterModeCommand(True), 'y', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:36.633) - winInputHook (12660):
Input: kb(laptop):space
IO - speech.speech.speak (14:14:36.649) - MainThread (2936):
Speaking ['space', EndUtteranceCommand()]
IO - speech.speech.speak (14:14:36.650) - MainThread (2936):
Speaking ['pressed']
DEBUG - gui.message.MessageDialog._onButtonEvent (14:14:36.726) - MainThread (2936):
Got button event on id=5103
DEBUG - gui.message.MessageDialog._onCloseEvent (14:14:36.736) - MainThread (2936):
Queueing <gui.message.MessageDialog object at 0x040ABFA0> for destruction
DEBUG - gui.message.MessageDialog._onCloseEvent (14:14:36.736) - MainThread (2936):
Removing <gui.message.MessageDialog object at 0x040ABFA0> from instances.
INFO - _remoteClient.client.RemoteClient.disconnect (14:14:36.736) - MainThread (2936):
Disconnecting from remote session
INFO - _remoteClient.secureDesktop.SecureDesktopHandler.followerSession (14:14:36.737) - MainThread (2936):
Updating follower session reference
DEBUG - _remoteClient.transport.Transport.unregisterInbound (14:14:36.737) - MainThread (2936):
Unregistered handler for set_braille_info
INFO - _remoteClient.client.onDisconnectedAsFollower (14:14:36.844) - MainThread (2936):
Control connector disconnected
IO - speech.speech.speak (14:14:36.854) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:36.859) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
IO - speech.speech.speak (14:14:36.865) - MainThread (2936):
Speaking ['Disconnected']
IO - inputCore.InputManager.executeGesture (14:14:37.639) - winInputHook (12660):
Input: kb(laptop):NVDA+n
DEBUG - gui.SysTrayIcon.evaluateUpdatePendingUpdateMenuItemCommand (14:14:37.641) - MainThread (2936):
Error while removing pending update menu item
Traceback (most recent call last):
  File "gui\__init__.pyc", line 720, in evaluateUpdatePendingUpdateMenuItemCommand
wx._core.wxAssertionError: C++ assertion ""node"" failed at ..\..\src\common\menucmn.cpp(462) in wxMenuBase::Remove(): removing item not in the menu?
IO - speech.speech.speak (14:14:37.669) - MainThread (2936):
Speaking ['NVDA', 'menu', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:37.857) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:37.863) - MainThread (2936):
Speaking ['Preferences', 'subMenu', CharacterModeCommand(True), 'p', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:38.004) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:38.009) - MainThread (2936):
Speaking ['Tools', 'subMenu', CharacterModeCommand(True), 't', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:38.345) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:38.359) - MainThread (2936):
Speaking ['View log', CharacterModeCommand(True), 'l', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:38.679) - winInputHook (12660):
Input: kb(laptop):upArrow
IO - speech.speech.speak (14:14:38.685) - MainThread (2936):
Speaking ['Remote Access', 'subMenu', CharacterModeCommand(True), 'e', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:39.094) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:39.106) - MainThread (2936):
Speaking ['Connect...', CharacterModeCommand(True), 'c', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:39.789) - winInputHook (12660):
Input: kb(laptop):enter
DEBUG - gui.contextHelp.bindHelpEvent (14:14:39.796) - MainThread (2936):
Did context help binding for DirectConnectDialog
DEBUG - gui.contextHelp.bindHelpEvent (14:14:39.812) - MainThread (2936):
Did context help binding for ClientPanel
DEBUG - gui.contextHelp.bindHelpEvent (14:14:39.825) - MainThread (2936):
Did context help binding for ServerPanel
IO - speech.speech.speak (14:14:39.956) - MainThread (2936):
Speaking ['Connect to Another Computer', 'dialog', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:39.962) - MainThread (2936):
Speaking ['Mode:', 'combo box', 'Allow this computer to be controlled', 'collapsed', 'Alt+', CharacterModeCommand(True), 'm', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:40.891) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:40.905) - MainThread (2936):
Speaking ['Server:', 'combo box', 'Use existing', 'collapsed', 'Alt+', CharacterModeCommand(True), 's', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:41.414) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:41.442) - MainThread (2936):
Speaking ['Host:', 'combo box', 'collapsed', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:41.443) - MainThread (2936):
Speaking ['edit', 'Alt+', CharacterModeCommand(True), 'h', CharacterModeCommand(False), CancellableSpeech (still valid), 'selected myownserver.com:6837']
IO - inputCore.InputManager.executeGesture (14:14:41.640) - winInputHook (12660):
Input: kb(laptop):tab
DEBUG - NVDAObjects.NVDAObject._get_placeholder (14:14:41.656) - MainThread (2936):
Potential unimplemented child class: <NVDAObjects.Dynamic_IAccessibleEditWindowNVDAObject object at 0x09923D50>
IO - speech.speech.speak (14:14:41.656) - MainThread (2936):
Speaking ['Key:', 'edit', 'Alt+', CharacterModeCommand(True), 'k', CharacterModeCommand(False), CancellableSpeech (still valid), 'blank']
INFO - _remoteClient.transport.ConnectorThread.run (14:14:41.739) - Thread-65_connector_loop (14556):
Ending control connector thread Thread-65_connector_loop
IO - inputCore.InputManager.executeGesture (14:14:41.821) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:41.833) - MainThread (2936):
Speaking ['Generate Key', 'button', 'Alt+', CharacterModeCommand(True), 'g', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:42.012) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:42.028) - MainThread (2936):
Speaking ['OK', 'button', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:42.210) - winInputHook (12660):
Input: kb(laptop):tab
IO - speech.speech.speak (14:14:42.218) - MainThread (2936):
Speaking ['Cancel', 'button', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:42.583) - winInputHook (12660):
Input: kb(laptop):space
IO - speech.speech.speak (14:14:42.614) - MainThread (2936):
Speaking ['space', EndUtteranceCommand()]
IO - speech.speech.speak (14:14:42.615) - MainThread (2936):
Speaking ['pressed']
IO - speech.speech.speak (14:14:42.722) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:42.728) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
IO - inputCore.InputManager.executeGesture (14:14:43.039) - winInputHook (12660):
Input: kb(laptop):NVDA+n
DEBUG - gui.SysTrayIcon.evaluateUpdatePendingUpdateMenuItemCommand (14:14:43.043) - MainThread (2936):
Error while removing pending update menu item
Traceback (most recent call last):
  File "gui\__init__.pyc", line 720, in evaluateUpdatePendingUpdateMenuItemCommand
wx._core.wxAssertionError: C++ assertion ""node"" failed at ..\..\src\common\menucmn.cpp(462) in wxMenuBase::Remove(): removing item not in the menu?
IO - speech.speech.speak (14:14:43.073) - MainThread (2936):
Speaking ['NVDA', 'menu', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:43.229) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:43.235) - MainThread (2936):
Speaking ['Preferences', 'subMenu', CharacterModeCommand(True), 'p', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:43.403) - winInputHook (12660):
Input: kb(laptop):downArrow
IO - speech.speech.speak (14:14:43.410) - MainThread (2936):
Speaking ['Tools', 'subMenu', CharacterModeCommand(True), 't', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:43.608) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:43.621) - MainThread (2936):
Speaking ['View log', CharacterModeCommand(True), 'l', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:43.832) - winInputHook (12660):
Input: kb(laptop):upArrow
IO - speech.speech.speak (14:14:43.839) - MainThread (2936):
Speaking ['Remote Access', 'subMenu', CharacterModeCommand(True), 'e', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:44.091) - winInputHook (12660):
Input: kb(laptop):enter
IO - speech.speech.speak (14:14:44.106) - MainThread (2936):
Speaking ['Connect...', CharacterModeCommand(True), 'c', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:44.739) - winInputHook (12660):
Input: kb(laptop):escape
IO - speech.speech.speak (14:14:44.749) - MainThread (2936):
Speaking ['Remote Access', 'subMenu', CharacterModeCommand(True), 'e', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:45.458) - winInputHook (12660):
Input: kb(laptop):escape
IO - speech.speech.speak (14:14:45.466) - MainThread (2936):
Speaking ['Tools', 'subMenu', CharacterModeCommand(True), 't', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:46.183) - winInputHook (12660):
Input: kb(laptop):escape
IO - speech.speech.speak (14:14:46.224) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:46.246) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
IO - speech.speech.speak (14:14:47.721) - MainThread (2936):
Speaking ['File', 'subMenu', 'Alt+', CharacterModeCommand(True), 'f', CharacterModeCommand(False), CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (14:14:48.581) - winInputHook (12660):
Input: kb(laptop):escape
IO - speech.speech.speak (14:14:48.698) - MainThread (2936):
Speaking ['new 4 - Notepad++', CancellableSpeech (still valid)]
IO - speech.speech.speak (14:14:48.706) - MainThread (2936):
Speaking ['edit', 'multi line', CancellableSpeech (still valid), 'blank']
```

### System Configuration
#### NVDA Installed/Portable/Running From Source
Installed

#### NVDA Version
2025.2beta1 (2025.2.0.37080)

#### Windows Version
Windows 11 24H2 (AMD64) build 26100.4484

#### Name and Version of Other Software in Use When Reproducing the Issue
None

#### Other Information About Your System
None

### Other questions
#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. This behaviors exist since this feature was introduced.

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

--- a/source/_remoteClient/client.py
+++ b/source/_remoteClient/client.py
@@ -248,6 +248,7 @@ class RemoteClient:
 			log.debug("Disconnect called but no active sessions")
 			return
 		log.info("Disconnecting from remote session")
+		self.connecting = False
 		if self.localControlServer is not None:
 			self.localControlServer.close()
 			self.localControlServer = None
@@ -263,6 +264,7 @@ class RemoteClient:
 		self.leaderSession.close()
 		self.leaderSession = None
 		self.leaderTransport = None
+		self.connecting = False
 
 	def disconnectAsFollower(self):
 		"""Close follower session and clean up related resources."""
@@ -270,6 +272,7 @@ class RemoteClient:
 		self.followerSession = None
 		self.followerTransport = None
 		self.sdHandler.followerSession = None
+		self.connecting = False
 
 	@alwaysCallAfter
 	def onConnectAsLeaderFailed(self):

--- a/source/_remoteClient/client.py
+++ b/source/_remoteClient/client.py
@@ -46,7 +46,7 @@ class RemoteClient:
 	followerSession: Optional[FollowerSession]
 	keyModifiers: Set[KeyModifier]
 	hostPendingModifiers: Set[KeyModifier]
-	connecting: bool
+	_connecting: bool
 	leaderTransport: Optional[RelayTransport]
 	followerTransport: Optional[RelayTransport]
 	localControlServer: Optional[server.LocalRelayServer]
@@ -65,7 +65,7 @@ class RemoteClient:
 		self.menu: Optional[RemoteMenu] = None
 		if not isRunningOnSecureDesktop():
 			self.menu: Optional[RemoteMenu] = RemoteMenu(self)
-		self.connecting = False
+		self._connecting = False
 		urlHandler.registerURLHandler()
 		self.leaderTransport = None
 		self.followerTransport = None
@@ -200,7 +200,7 @@ class RemoteClient:
 		log.info(
 			f"Initiating connection as {connectionInfo.mode} to {connectionInfo.hostname}:{connectionInfo.port}",
 		)
-		self.connecting = True
+		self._connecting = True
 		if connectionInfo.mode == ConnectionMode.LEADER:
 			self.connectAsLeader(connectionInfo)
 		elif connectionInfo.mode == ConnectionMode.FOLLOWER:
@@ -248,7 +248,7 @@ class RemoteClient:
 			log.debug("Disconnect called but no active sessions")
 			return
 		log.info("Disconnecting from remote session")
-		self.connecting = False
+		self._connecting = False
 		if self.localControlServer is not None:
 			self.localControlServer.close()
 			self.localControlServer = None
@@ -264,7 +264,7 @@ class RemoteClient:
 		self.leaderSession.close()
 		self.leaderSession = None
 		self.leaderTransport = None
-		self.connecting = False
+		self._connecting = False
 
 	def disconnectAsFollower(self):
 		"""Close follower session and clean up related resources."""
@@ -272,7 +272,7 @@ class RemoteClient:
 		self.followerSession = None
 		self.followerTransport = None
 		self.sdHandler.followerSession = None
-		self.connecting = False
+		self._connecting = False
 
 	@alwaysCallAfter
 	def onConnectAsLeaderFailed(self):
@@ -303,7 +303,7 @@ class RemoteClient:
 				stack_info=True,
 			)
 			return
-		if self.isConnected() or self.connecting:
+		if self.isConnected() or self._connecting:
 			gui.messageBox(
 				pgettext(
 					"remote",
@@ -360,7 +360,7 @@ class RemoteClient:
 	@alwaysCallAfter
 	def onConnectedAsLeader(self):
 		log.info("Successfully connected as leader")
-		self.connecting = False
+		self._connecting = False
 		configuration.writeConnectionToConfig(self.leaderSession.getConnectionInfo())
 		if self.menu:
 			self.menu.handleConnected(ConnectionMode.LEADER, True)
@@ -409,7 +409,7 @@ class RemoteClient:
 	@alwaysCallAfter
 	def onConnectedAsFollower(self):
 		log.info("Control connector connected")
-		self.connecting = False
+		self._connecting = False
 		cues.controlServerConnected()
 		if self.menu:
 			self.menu.handleConnected(ConnectionMode.FOLLOWER, True)
@@ -628,7 +628,7 @@ class RemoteClient:
 		:note: Shows confirmation dialog before connecting
 		:raises: Displays error if already connected
 		"""
-		if self.isConnected() or self.connecting:
+		if self.isConnected() or self._connecting:
 			gui.messageBox(
 				pgettext(
 					"remote",
@@ -641,7 +641,7 @@ class RemoteClient:
 			)
 			return
 
-		self.connecting = True
+		self._connecting = True
 		try:
 			serverAddr = conInfo.getAddress()
 			key = conInfo.key
@@ -676,7 +676,7 @@ class RemoteClient:
 			):
 				self.connect(conInfo)
 		finally:
-			self.connecting = False
+			self._connecting = False
 
 	def isConnected(self) -> bool:
 		"""Check if there is an active connection.

--- a/source/_remoteClient/client.py
+++ b/source/_remoteClient/client.py
@@ -303,7 +303,7 @@ class RemoteClient:
 				stack_info=True,
 			)
 			return
-		if self.isConnected() or self._connecting:
+		if self.isConnected() or self.isConnecting:
 			gui.messageBox(
 				pgettext(
 					"remote",
@@ -628,7 +628,7 @@ class RemoteClient:
 		:note: Shows confirmation dialog before connecting
 		:raises: Displays error if already connected
 		"""
-		if self.isConnected() or self._connecting:
+		if self.isConnected() or self.isConnecting:
 			gui.messageBox(
 				pgettext(
 					"remote",
@@ -678,13 +678,16 @@ class RemoteClient:
 		finally:
 			self._connecting = False
 
+	@property
+	def _transport(self) -> RelayTransport | None:
+		return self.followerTransport or self.leaderTransport
+
 	def isConnected(self) -> bool:
 		"""Check if there is an active connection.
 
 		:return: True if either follower or leader transport is connected
 		"""
-		connector = self.followerTransport or self.leaderTransport
-		if connector is not None:
+		if (connector := self._transport) is not None:
 			return connector.connected
 		return False
 
@@ -694,6 +697,15 @@ class RemoteClient:
 		if self.followerTransport is not None:
 			return self.followerTransport.connected
 		return False
+
+	@property
+	def isConnecting(self) -> bool:
+		"""Whether or not a session is being connected or reconnected to."""
+		return self._connecting or (
+			not self.isConnected()
+			and self._transport is not None
+			and self._transport.reconnectorThread.running
+		)
 
 	def registerLocalScript(self, script: scriptHandler._ScriptFunctionT):
 		"""Add a script to be handled locally instead of sent to remote.

--- a/source/_remoteClient/client.py
+++ b/source/_remoteClient/client.py
@@ -264,6 +264,8 @@ class RemoteClient:
 		self.leaderSession.close()
 		self.leaderSession = None
 		self.leaderTransport = None
+		if self.menu:
+			self.menu.handleConnected(ConnectionMode.LEADER, False)
 		self._connecting = False
 
 	def disconnectAsFollower(self):
@@ -272,6 +274,8 @@ class RemoteClient:
 		self.followerSession = None
 		self.followerTransport = None
 		self.sdHandler.followerSession = None
+		if self.menu:
+			self.menu.handleConnected(ConnectionMode.FOLLOWER, False)
 		self._connecting = False
 
 	@alwaysCallAfter

--- a/source/_remoteClient/client.py
+++ b/source/_remoteClient/client.py
@@ -200,6 +200,7 @@ class RemoteClient:
 		log.info(
 			f"Initiating connection as {connectionInfo.mode} to {connectionInfo.hostname}:{connectionInfo.port}",
 		)
+		self.connecting = True
 		if connectionInfo.mode == ConnectionMode.LEADER:
 			self.connectAsLeader(connectionInfo)
 		elif connectionInfo.mode == ConnectionMode.FOLLOWER:
@@ -299,6 +300,19 @@ class RemoteClient:
 				stack_info=True,
 			)
 			return
+		if self.isConnected() or self.connecting:
+			gui.messageBox(
+				pgettext(
+					"remote",
+					# Translators: Message shown when trying to connect while already connected.
+					"A Remote Access session is already in progress. Disconnect before starting a new session.",
+				),
+				# Translators: Title of the connection error dialog.
+				pgettext("remote", "Already Connected"),
+				wx.OK | wx.ICON_WARNING,
+			)
+			return
+
 		previousConnections = configuration.getRemoteConfig()["connections"]["lastConnected"]
 		hostnames = list(reversed(previousConnections))
 		dlg = dialogs.DirectConnectDialog(
@@ -343,6 +357,7 @@ class RemoteClient:
 	@alwaysCallAfter
 	def onConnectedAsLeader(self):
 		log.info("Successfully connected as leader")
+		self.connecting = False
 		configuration.writeConnectionToConfig(self.leaderSession.getConnectionInfo())
 		if self.menu:
 			self.menu.handleConnected(ConnectionMode.LEADER, True)
@@ -391,6 +406,7 @@ class RemoteClient:
 	@alwaysCallAfter
 	def onConnectedAsFollower(self):
 		log.info("Control connector connected")
+		self.connecting = False
 		cues.controlServerConnected()
 		if self.menu:
 			self.menu.handleConnected(ConnectionMode.FOLLOWER, True)

--- a/source/globalCommands.py
+++ b/source/globalCommands.py
@@ -5020,7 +5020,7 @@ class GlobalCommands(ScriptableObject):
 		gui.blockAction.Context.MODAL_DIALOG_OPEN,
 	)
 	def script_connectToRemote(self, gesture: "inputCore.InputGesture"):
-		if _remoteClient._remoteClient.isConnected() or _remoteClient._remoteClient.connecting:
+		if _remoteClient._remoteClient.isConnected() or _remoteClient._remoteClient.isConnecting:
 			# Translators: A message indicating that the remote client is already connected.
 			ui.message(_("Already connected"))
 			return

--- a/source/globalCommands.py
+++ b/source/globalCommands.py
@@ -5034,7 +5034,7 @@ class GlobalCommands(ScriptableObject):
 	)
 	@gui.blockAction.when(gui.blockAction.Context.REMOTE_ACCESS_DISABLED)
 	def script_toggleRemoteConnection(self, gesture: "inputCore.InputGesture") -> None:
-		if _remoteClient._remoteClient.isConnected():
+		if _remoteClient._remoteClient.isConnected() or _remoteClient._remoteClient.isConnecting:
 			self.script_disconnectFromRemote(gesture)
 		else:
 			self.script_connectToRemote(gesture)

--- a/tests/unit/test_remote/test_remoteClient.py
+++ b/tests/unit/test_remote/test_remoteClient.py
@@ -31,6 +31,8 @@ class FakeMenu:
 		def Check(self, value):
 			self.checked = value
 
+	def handleConnected(self, *args, **kwargs): ...
+
 
 class FakeTransport:
 	def __init__(self):

--- a/user_docs/en/changes.md
+++ b/user_docs/en/changes.md
@@ -31,6 +31,7 @@
 * Fixed a problem where NVDA fails to start with an SAPI5 Eloquence voice and falls back to OneCore voices. (#18301, @gexgd0419)
 * Fixed Highlighter not working with Outlook contact auto-complete lists. (#18483, @Nerlant)
 * Fixed a bug which stopped speech from working via NVDA Remote Access when the controlled computer had no audio output devices enabled. (#18544)
+* Fixed a bug which caused NVDA Remote Access to stop working if a session was interrupted while connecting to the server. (#18476)
 
 ### Changes for Developers
 

